mesonbuild/wrap/wrap.py

```diff
@@ -422,13 +418,22 @@ def get_file_internal(self, what: str) -> str:
             return path.as_posix()
 
     def apply_patch(self) -> None:
-        path = self.get_file_internal('patch')
-        try:
-            shutil.unpack_archive(path, self.subdir_root)
-        except Exception:
-            with tempfile.TemporaryDirectory() as workdir:
-                shutil.unpack_archive(path, workdir)
-                self.copy_tree(workdir, self.subdir_root)
+        if 'patch_filename' in self.wrap.values:
+            path = self.get_file_internal('patch')
```

Comment:
This should check if `patch_directory` is also defined and error out if it is. Otherwise the file is ambiguous.

Reply:
Good point, fixed now.